chat-plugins/animesearch.js

```diff
@@ -89,9 +79,15 @@ exports.commands = {
 		if (this.broadcasting && room.id === 'lobby') return this.errorReply("This command cannot be broadcasted in the Lobby.");
 		if (!target || !target.trim()) return this.sendReply('/' + cmd + ' [query] - Searches for an anime based on the given search query.');
 
-		search(target.trim()).then(function (response) {
-			this.sendReplyBox(response);
-			if (this.broadcasting) room.update();
-		}.bind(this));
+		searchAnime(target.trim()).then((anime) => {
+			let genres = anime.genres.map(function (genre) { return toId(genre.name) });
+			if (!room.isPrivate && (genres.indexOf('hentai') > -1 || (genres.indexOf('yaoi') > -1 && anime.title.match(/pico/ig))) && this.broadcasting)
```

Comment:
Let's just prevent hentai from being able to be searched with this entire script regardless of room, please.